examples/src/walkable-path.ts

```diff
@@ -0,0 +1,151 @@
+/*!
+ * Copyright 2020 Cognite AS
+ */
+
+import * as THREE from 'three';
+import CameraControls from 'camera-controls';
+import * as reveal from '@cognite/reveal';
+import { CogniteClient } from '@cognite/sdk';
+import { toThreeVector3, CadNode } from '@cognite/reveal/threejs';
+import { vec3 } from 'gl-matrix';
+import { SectorModelTransformation } from '@cognite/reveal/models/cad/types';
+
+CameraControls.install({ THREE });
+
+interface TransitPathResponse {
+  items: [
+    {
+      segments: [
+        {
+          path: [
+            {
+              x: number;
+              y: number;
+              z: number;
+            }
+          ];
+        }
+      ];
+    }
+  ];
+}
+
+interface TransitPathRequest {
+  modelId: number;
+  items: [
+    {
+      from: { x: number; y: number; z: number } | { nodeId: number };
+      to: { x: number; y: number; z: number } | { nodeId: number };
+      movingObjectSize: { diameter: number; height: number };
+    }
+  ];
+}
+
+// Should find a way to add this to maybe cognite-sdk-js later?
+async function getTransitPath(client: CogniteClient, transitRequest: TransitPathRequest) {
+  const project = '3ddemo';
+  const url = `https://api.cognitedata.com/api/playground/projects/${project}/3d/pathfinder/transit`;
+  const response = await client.post<TransitPathResponse>(url, {
+    data: transitRequest
+  });
+  return response;
+}
+
+// TODO: Should be added to reveal/viewer. The input data TransitPath response is in my
+// opinion too coupled with our sdk response, should we split it into two funcitons, one taking in a { x, y, z}[]
+// and another converting the response to said array.
+
+function createWalkablePath(pointData: TransitPathResponse, modelTransformation: SectorModelTransformation) {
+  const pathVectors: THREE.Vector3[] = [];
+  const vector: vec3 = vec3.create();
+  for (const item of pointData.items) {
+    for (const segment of item.segments) {
+      for (const path of segment.path) {
+        vec3.set(vector, path.x, path.y, path.z);
+        pathVectors.push(toThreeVector3(new THREE.Vector3(), vector, modelTransformation));
+      }
+    }
+  }
+  return pathVectors;
+}
+
+// TODO: Add to some kind of primitives like class in reveal viewer?
+
+function createPathGeometry(
+  points: THREE.Vector3[],
+  segments: number,
+  radius: number,
+  radiusSegments: number,
+  closed: boolean
+) {
+  const curve = new THREE.CatmullRomCurve3(points);
+  const geometry = new THREE.TubeBufferGeometry(curve, segments, radius, radiusSegments, closed);
+  return geometry;
+}
+
+async function main() {
+  const client: CogniteClient = new CogniteClient({ appId: 'WalkablePath Example' });
+  client.loginWithOAuth({
+    project: '3ddemo'
```

Comment:
Same as Lars' comment below: 3ddemo is not public and should not be hard-coded in the examples. I do not think we have any public tenant that can be used, so rather add a dialog and/or use URL parameters.

Reply:
Url parameters it is.